two_body_scf_guess_III.py

```diff
@@ -0,0 +1,146 @@
+from wrap_pyscf import *
+import numpy as np
+from itertools import combinations
+
+def make_monomer_dens(monomers, bs):
+    monomer_dens = []
+    for j in monomers:
+        k = density(j, bs)
+        monomer_dens.append(k)
+    return monomer_dens
+
+def make_offset(a):
+    lst = []
+    lst.append(a-1)
+    for i, num in enumerate(lst):
+        if num > 1:
+            k = lst[-1]-1
+            lst.append(k)
+    lst1 = lst
+    count = 0
+    lst2 = []
+    for item in lst1:
+        count = count + item
+        lst2.append(count)
+    offset = lst2
+    return offset
+
+def dimer_lst(monomers, a, unit_guess):
+    mono_com = list(combinations(monomers, unit_guess))
+    lst = [[] for _ in range(a-1)]
+    for i, m_c in enumerate(mono_com):
+        (x, y) = mono_com[i]
+        k = x+y
+        lst.append(k)
+    dimers = lst
+    return dimers
+
+def sorted_dimers(unsorted_dimers, offset, a):
+    for i, lst in enumerate(unsorted_dimers):
+        for j, off in enumerate(offset):
+            if i <= (a-2):
+                if len(lst) <= (a-2)-i:
+                    unsorted_dimers[i].append(unsorted_dimers[(i+j)+offset[i]])
+    del unsorted_dimers[(a-1):]
+    s_dimers = unsorted_dimers
+    return s_dimers
+
+def make_dimers(monomers, unit_guess):
+    a = len(monomers)
+    offset = make_offset(a)
+    unsorted_dimers = dimer_lst(monomers, a, unit_guess)
+    s_dimers = sorted_dimers(unsorted_dimers, offset, a)
+    return s_dimers
```

Comment:
Did you want this function to make a dimer dictionary of the format you already described? It currently makes an list of lists, where each monomer-ordered inner list contains unique (no repeating) dimers. E.g. this is what it looks like for a trimer: `[[AB, AC], [BC]]`. If not I can make so that when `two-body-scf-guess-III` is executed, a dictionary of the dimers is printed.